crates/oxc_semantic/src/scope.rs

```diff
@@ -38,7 +39,23 @@ impl ScopeTree {
         std::iter::successors(Some(scope_id), |scope_id| self.parent_ids[*scope_id])
     }
 
-    pub fn descendants(&self) -> impl Iterator<Item = ScopeId> + '_ {
+    pub fn descendants(&self, scope_id: ScopeId) -> impl Iterator<Item = ScopeId> + '_ {
+        let mut list = vec![];
+        let mut queue = VecDeque::from_iter([scope_id]);
+
+        while let Some(current_id) = queue.pop_front() {
```

Comment:
This algo's time complexity could be `O(n^2)`

Reply:
We could create a new field called `child_ids` like `parent_ids` , it store each direct child scopeids, then `descendants` could just be a linear scan.

Reply:
Let me give that a shot.